html/beps/bep_0046.rst

```diff
@@ -0,0 +1,238 @@
+:BEP: 46
+:Title: Have message for multiple pieces
+:Version: $Revision$
+:Last-Modified: $Date$
+:Author:  Arvid Norberg <arvid@libtorrent.org>
+:Status:  Draft
+:Type:    Standards Track
+:Content-Type: text/x-rst
+:Created: 26-May-2016
+:Post-History: 
+
+
+Abstract
+========
+
+This BEP proposes an extension that aims to significantly reduce the overhead
+of torrents with many pieces, by making representations more compact and
+enabling coalescing of HAVE messages.
+
+Rationale
+=========
+
+Small pieces have significant benefits. They enable:
+
+1. Shorter delays from downloading data until it can be
+   reliably consumed (important for streaming)
+2. Shorter delays forwarding downloaded bytes to other peers. This is most
+   obvious when first joining a swarm, but also significant in swarms
+   bottlenecked by its seeds
+
+Pieces should be small, ideally 16 kiB. The costs associated with small pieces
+are:
+
+1. the size of the .torrent file. Each piece hash is 20 bytes. a 4GiB torrent
+   with 16 kiB pieces would have a .torrent file greater than 5 MB.
+2. The size of bitfield messages. Every peer that has some but not all pieces
+   send a bitfield during peer handshake. With this hypothetical 4 GiB torrent,
+   each bitfield message would be 32 kiB (equivalent to two payload blocks).
+3. The number of HAVE messages that would need to be sent to peers during
+   downloads would be a lot higher. Each HAVE message is 9 bytes (length prefix,
+   type and piece index) and possibly another 20 bytes if sent on an otherwise
+	idle connection (TCP/IP overhead). Compared to the same torrent with a piece
+   size common today (2 MiB), it would require 128 times as many messages.
+
+Issue (1) is addressed by merkle tree torrents, in BEP30_.
+
+.. _BEP30: http://bittorrent.org/beps/bep_0030.html
+
+This BEP attempts to address (2) and (3) with a few different extensions.
+
+Redundant HAVEs
```

Comment:
I feel pretty confident the redundant HAVE portion is non-controversial and a clear benefit.